packages/python/plotly/plotly/express/_core.py

```diff
@@ -434,11 +442,12 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
         mapping_labels_copy = OrderedDict(mapping_labels)
         if args["hover_data"] and isinstance(args["hover_data"], dict):
             for k, v in mapping_labels.items():
-                if k in args["hover_data"]:
-                    if args["hover_data"][k][0]:
-                        if isinstance(args["hover_data"][k][0], str):
+                k_args = invert_label(args, k)
```

Comment:
OK so the problem is that `k` here is the renamed label? so if `labels=dict(a="A")` then `k` would be `"A"` and we need to find `a`?

Reply:
might want to add a comment here explaining... it took me 4 tries and 10min to figure this one out :)

Reply:
ok :-). Yes, that's what you said.